File: python_implemented/linearSearch.py
#Linear search

def linearSearch(arr, x):
	count = 0
	for i in arr:
		count += 1
		if i == x:
			return count
	if count == n:
		return -1

def main():
	var = raw_input("Enter the elements: ")
	arr = map(int, var.split())
	x = input("Enter search item: ")
	
	pos = linearSearch(arr, x)

	if pos > 0:
		print "Position of %d is %d.\n" % (x, pos)
	else:
		print "Element not found.\n"

main()

File: C-C++/linearSearch.cpp
#include <iostream>
#include<conio.h>
using namespace std;

int main()
{
	int n, x;
	cout<<"Enter array size:";
	cin>>n;
	int a[n];
	cout<<"Enter elements to array:";

	for(int i = 0; i < n; i++)
		cin>>a[i];
	cout<<"Enter number to search:";
	cin>>x;

	for(int i = 0; i < n; i++)
	{
		if(x < a[0] || x > a[n-1])
		{
			cout<<"<Not present>"<<endl;
			break;
		}
		else
		{
			if(x == a[i])
			{
				cout<<"Found at pos: "<<i+1<<endl;
				break;
			}
		}
	}
	return 0;
}
